### PR TITLE
add ci cleanup

### PR DESCRIPTION
**Pull request by seethroughdev**
This PR addresses adding eslint and prettier to the build. 

Now that we have a couple of people contributing, it will be helpful to make sure we get some of these checks in before PR. That way we can focus on the code itself and not formatting or lint issues!

It will run automatically on `build`. or you can run:

`npm run lint/format/prebuild` to get the output.

### CODE REVIEW

**Comment by seethroughdev** [comment]
@Lanjelin , the initial cleanup touched a bunch of your code, so i want to make sure it didn't change the intent of anything you wrote. Could you carefully review the changes? And maybe add to your test and make sure it didn't create any unexpected results?

Sent you the invite for collaborator, once you accept, i'll put you on as a reviewer. 

Thanks!

File: src/main.ts [hunk 1]
```diff
@@ -215,7 +220,7 @@ export default class RecipeGrabber extends Plugin {
 				// magicTime datePublished
 			} else if (arguments.length == 3) {
 				// magicTime datePublished "dd-mm-yyyy HH:MM"
-				if (new Date(arg1) == "Invalid Date") {
```
Comment (seethroughdev) [inline src/main.ts:218]: this check won't ever return true. So, added a `isValidDate` function above, so it can be shared.

File: src/main.ts [hunk 2]
```diff
@@ -280,15 +285,19 @@ export default class RecipeGrabber extends Plugin {
...
 			// pages can have multiple recipes, lets add them all
-			let i = 0;
 			for (const recipe of recipes) {
 				if (this.settings.debug) {
 					console.log(recipe);
 					console.log(markdown(recipe));
 				}
 				// this will download the images and replace the json "recipe.image" value with the path of the image file.
 				if (this.settings.saveImg && file) {
-					const filename = recipe.name.replace(/\ /g, "-"); // We dont want spaces in filenames
```
Comment (seethroughdev) [inline src/main.ts:291]: had some type issues here, also, current is replacing all spaces with `-`. thinking you wanted any consecutive spaces with a single dash.

File: src/main.ts [hunk 3]
```diff
@@ -406,15 +420,16 @@ export default class RecipeGrabber extends Plugin {
 		filename: Recipe["name"],
 		imgUrl: Recipe["image"],
 		file: TFile,
-		imgNum = false,
```
Comment (seethroughdev) [inline src/main.ts:409]: eslint wants us to keep value as number, here it was being used as both number and bool

File: src/main.ts [hunk 4]
```diff
@@ -424,25 +439,23 @@ export default class RecipeGrabber extends Plugin {
 			if (!type) {
 				return false;
 			}
-			let path;
+			let path = "";
 			if (this.settings.imgFolder == "") {
-				//@ts-ignore
```
Comment (seethroughdev) [inline src/main.ts:429]: eslint didn't want any use of `ts-ignore`. but thinking the `any` fix here should do the trick.

File: src/main.ts [hunk 5]
```diff
@@ -175,25 +175,31 @@ export default class RecipeGrabber extends Plugin {
...
 			// Handlebars appends an ubject to the arguments
-			if (arguments.length == 1) {
+			if (arguments.length === 1) {
 				// magicTime
 				return dateFormat(new Date(), "yyyy-mm-dd HH:MM");
-			} else if (arguments.length == 2) {
-				if (new Date(arg1) == "Invalid Date") {
+			} else if (arguments.length === 2) {
+				// check if arg1 is a valid date
+				if (isValidDate(arg1)) {
```
Comment (Lanjelin) [inline src/main.ts:202]:
```suggestion
				if (!isValidDate(arg1)) {
```
If it's not a valid date, check if it's a ISO 8601 duration string (PT1H30M), else assume it's a valid dateFormat mask.

File: src/main.ts [hunk 6]
```diff
@@ -215,7 +220,7 @@ export default class RecipeGrabber extends Plugin {
 				// magicTime datePublished
 			} else if (arguments.length == 3) {
 				// magicTime datePublished "dd-mm-yyyy HH:MM"
-				if (new Date(arg1) == "Invalid Date") {
+				if (isValidDate(arg1)) {
```
Comment (Lanjelin) [inline src/main.ts:223]:
```suggestion
				if (!isValidDate(arg1)) {
```
If it's not a valid date, return an error.

File: src/main.ts [hunk 7]
```diff
@@ -215,7 +220,7 @@ export default class RecipeGrabber extends Plugin {
 				// magicTime datePublished
 			} else if (arguments.length == 3) {
```
Comment (Lanjelin) [inline src/main.ts:221]:
```suggestion
			} else if (arguments.length === 3) {
```

File: src/main.ts (hunk 6 above)
Comment (seethroughdev) [reply src/main.ts:223]: Ha. Good catch!

File: src/main.ts (hunk 1 above)
Comment (seethroughdev) [reply src/main.ts:218]: Wasn't sure how you wanted to fix the reorder, do you want to pull the branch, or you can just make the suggested change, and i can merge.

File: src/main.ts (hunk 2 above)
Comment (seethroughdev) [reply src/main.ts:291]: Yep. We could easily pull in a library to do this right, but i think what you have is fine for most use cases.

Comment (seethroughdev) [reply src/main.ts:291]: My regex change only really adds multiple spaces with a single `-`

File: src/main.ts (hunk 5 above)
Comment (seethroughdev) [reply src/main.ts:202]: Can you add a suggested change to the new version or update the PR? not sure how you'd like it to check.

**Comment by Lanjelin** [comment]
Reordered the `isValidDate`-checks in `magicTime` for better readability.  
I've run the updated version manually through my list of recipes to check.

**Comment by seethroughdev** [comment]
Excellent. thanks again @Lanjelin . I'll merge in and try and release tomorrow!

**Pull request merged**